src/openrct2/paint/Paint.cpp

```diff
@@ -734,25 +732,25 @@ static void draw_pixel_info_crop_by_zoom(rct_drawpixelinfo* dpi)
 
 paint_session* paint_session_alloc(rct_drawpixelinfo* dpi, uint32_t viewFlags)
 {
-    _paintSessionsUsed++;
-    size_t index = _nextPaintSessionIndex.fetch_add(1);
-    if (index >= _paintSessions.size())
+    paint_session* session = nullptr;
+
+    if (_freePaintSessions.empty())
+    {
+        session = new paint_session();
```

Comment:
where's `delete`?

Reply:
No where, its supposed to stay until next use.

Reply:
I get that, but you're not releasing it after you're fully done with it.

Reply:
When would that be? Thats why I'm saying ideally this would belong to the Painter/Context/Renderer

Reply:
I would just leave it for now since its globally accessible, currently the lifetime ends with the process and the footprint is very minimal depending on resolution.

Reply:
It should be no later than reaching `main()`, earlier if possible.